indico/modules/users/controllers.py

```diff
@@ -852,6 +852,12 @@ class RHAcceptRegistrationRequest(RHRegistrationRequestBase):
     """Accept a registration request."""
 
     def _process(self):
+        user_exists = User.query.filter(User.all_emails == self.request.email).has_rows()
+        if user_exists:
```

Comment:
```suggestion
        if get_user_by_email(self.request.email):
```

That way you get the check for `~User.is_deleted` for free (deleted users may have colliding emails).

(You'll probably need to import it as well, I didn't check if this file is already importing it or not)

Reply:
Ah right, we don't enforce uniqueness of emails for deleted users.